### ISSUE TITLE
Hard to enable boot diag during VM create

### ISSUE THREAD

**Issue by JasonRShaver** [Azure/azure-cli#4059]
Unable to make a VM with boot time debugging (console logging) enabled as part of the create command.  Had to make the vm create async with --nowait  and then enable it immediately. (or that was the theory..  it didn't work) 

```
  az vm create --resource-group ${RESOURCEGROUP} --location ${LOCATION} \ 
    --name ${VMNAME} \
    --os-type linux \
     --attach-os-disk ${MANAGEDDISK} \
    --location $LOCATION \
     --size Standard_DS1_v2 \
    --no-wait  \   
    --ssh-key-value ${KEYFILE} \
    # Apparently not needed (?)   az vm boot-diagnostics enable /
    --storage "https://${STORAGEACCOUNTNAME}.blob.core.windows.net/"  /
    --name ${VMNAME} /
    --resource-group ${RESOURCEGROUP}
# need a way to now wait  for the VM to complete before we proceed.
# How do we wait?
```

That doesn't work BTW.. I would have liked a --enable_boot-diag   option in the create.  If you wait until the create is done, then it's too late because all the booting is finished.

**Comment by waldiTM** [comment]
Also the new serial console access stuff need boot diagnostics enabled.

**Comment by abudhu** [comment]
+1 on this.  There are certain marketplace items that write information to the system log on first boot and only first boot.  Enabling boot diagnostics after is too late via the CLI as mentioned in the original issue.

**Comment by yugangw-msft** [comment]
Fixed through #6302
